js/ui/calendar/ui.calendar.base_view.js

```diff
@@ -198,12 +205,29 @@ const BaseView = Widget.inherit({
                 });
             }
         }));
+
+        if(this.option('selectionMode') === 'range') {
+            this._createCellHoverAction();
+            eventsEngine.off(this._$table, CALENDAR_DXHOVER_EVENT_NAME);
+            eventsEngine.on(this._$table, CALENDAR_DXHOVER_EVENT_NAME, `td:not(.${CALENDAR_WEEK_NUMBER_CELL_CLASS})`, ((e) => {
```

Comment:
What do you think if we extract common selector `td:not(.${CALENDAR_WEEK_NUMBER_CELL_CLASS})` to const variable, it allows us to avoid duplication

Reply:
done